Development.md

```diff
@@ -0,0 +1,250 @@
+
+## Sandbox AWS Rack Install
+
+Rack consumes numerous AWS and Docker APIs. The easiest way to develop Rack is with real AWS access keys interacting with real AWS resources like a Dynamo Table, ECS Cluster, and CloudFormation Stack.
+
+This is easy to bootstrap with the Rack project itself:
+
+```
+$ convox install --stack-name dev
+```
+
+It can also be bootstrapped with no `convox` tools via the AWS CLI:
+
+```
+$ aws cloudformation create-stack --stack-name dev --template-body file://$(pwd)/api/dist/kernel.json
+```
+
+You can also use any existing Rack with the caveat that running a laptop Rack against it could have side effects like terminating instances.
+
+## AWS Rack Ingress
+
+Parts of the API like `convox ps --stats` interact with the Docker daemon running on every AWS instance. To enable `convox ps --stats` to work from a laptop you will want to open up access to the instance Docker daemons:
+
+* Open the [Security Group Management Console](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#SecurityGroups)
+* Select the Security Group with the Group Name like "dev-SecurityGroup-4PNOYR5HUH83" and the Description "Instances"
+* Click the Inbound tab, then the Edit button, then the Add Rule button
+* Keep Custom TCP Rule and TCP protocol
+* Add "2376" for Port Range, and select "My IP" for Source
+* Click the Save button
+
+Warning: Do not expose instance port 2376 to "Anywhere"! This will expose your Docker daemons to the whole Internet.
+
+## Rack Golang Project
+
+Rack is written in Golang. To setup a Go environment, see the excellent [Getting Started](https://golang.org/doc/install) docs. You can then clone and build the project with the `go get` tool:
+
+```
+$ go get github.com/convox/rack/...
+```
+
+After this, `which convox` should refer to `$GOPATH/bin/convox`.
+
+## Laptop Rack Environment
+
+The laptop Rack is running an API process that has AWS Access Keys, AWS resource names, and other various settings in its environment. You need to copy this to your laptop
+
+```
+$ cd $GOPATH/src/github.com/convox/rack
+
+# Introspect the dev rack to find the PID of the API web process
+
+$ STACK_NAME=$(convox api get /system | jq -r .name)
+$ WEB_PID=$(convox api get /apps/$STACK_NAME/processes | jq -r '.[] | select(.name == "web") | .id' | head -1)
+
+# Introspect the API web process to get its environment
+
+$ convox exec $WEB_PID env --app $STACK_NAME > .env
+```
+
+Now you have a bunch of secrets that will let your laptop interact with AWS APIs:
+
+```
+$ cat .env
+CLIENT_ID=noah@convox.com
+RACK=dev
+SUBNETS=subnet-13de3139,subnet-b5578fc3,subnet-21c13379
+AWS_ACCESS=AKIAIOSFODNN7EXAMPLE
+AWS_SECRET=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+AWS_REGION=us-east-1
+DYNAMO_RELEASES=dev-releases
+CLUSTER=dev-Cluster-1E4XJ0PQWNAYS
+PASSWORD=45e0f109-3f56-4b30-9b5a-b0939b8a4c25
+...
+```
+
+Note: `convox install --development` is deprecated in favor of introspecting with `convox env`.
```

Comment:
Fine with me. Removed the note and will open up an issue on rack to strip out some bits.